cmd/minikube/cmd/start.go

```diff
@@ -823,6 +823,10 @@ func selectImageRepository(mirrorCountry string, v semver.Version) (bool, string
 	for _, code := range tryCountries {
 		localRepos := constants.ImageRepositories[code]
 		for _, repo := range localRepos {
+			// Avoid network calls for tests
```

Comment:
instead of adding a new parameter to the func, consider using this better pattern to assign the func to a variagble, and for the unit test over rider that func with a different func
for example


https://github.com/medyagh/minikube/blob/721b3da7ecd20508a174a0d6f800813188fd948d/pkg/drivers/kic/oci/network_create_test.go#L26
```
var dockerInspectGetterMock = func(name string) (*RunResult, error) {
	var responseInBytes bytes.Buffer
	responseInBytes.WriteString(dockerResponse)
	response := &RunResult{Stdout: responseInBytes}

	return response, nil
}
```

and for the non-mock one use this

```

var dockerInsepctGetter = func(name string) (*RunResult, error) {
	// hack -- 'support ancient versions of docker again (template parsing issue) #10362' and resolve 'Template parsing error: template: :1: unexpected "=" in operand' / 'exit status 64'
	// note: docker v18.09.7 and older use go v1.10.8 and older, whereas support for '=' operator in go templates came in go v1.11
	cmd := exec.Command(Docker, "network", "inspect", name, "--format", `{"Name": "{{.Name}}","Driver": "{{.Driver}}","Subnet": "{{range .IPAM.Config}}{{.Subnet}}{{end}}","Gateway": "{{range .IPAM.Config}}{{.Gateway}}{{end}}","MTU": {{if (index .Options "com.docker.network.driver.mtu")}}{{(index .Options "com.docker.network.driver.mtu")}}{{else}}0{{end}}, "ContainerIPs": [{{range $k,$v := .Containers }}"{{$v.IPv4Address}}",{{end}}]}`)
	rr, err := runCmd(cmd)
	// remove extra ',' after the last element in the ContainerIPs slice
	rr.Stdout = *bytes.NewBuffer(bytes.ReplaceAll(rr.Stdout.Bytes(), []byte(",]"), []byte("]")))
	return rr, err
}
```
https://github.com/medyagh/minikube/blob/721b3da7ecd20508a174a0d6f800813188fd948d/pkg/drivers/kic/oci/network_create.go#L170

Reply:
Good call. Fixed.